pkg/storaged/swap/swap.jsx

```diff
@@ -27,14 +27,63 @@ import { useEvent } from "hooks";
 
 import { StorageCard, StorageDescription, new_card } from "../pages.jsx";
 import { format_dialog } from "../block/format-dialog.jsx";
-import { fmt_size, decode_filename } from "../utils.js";
+import {
+    fmt_size, decode_filename, encode_filename,
+    parse_options, unparse_options, extract_option,
+} from "../utils.js";
 import { std_lock_action } from "../crypto/actions.jsx";
 
 const _ = cockpit.gettext;
 
+async function set_swap_noauto(block, noauto) {
+    for (const conf of block.Configuration) {
+        if (conf[0] == "fstab") {
+            const options = parse_options(decode_filename(conf[1].opts.v));
+            extract_option(options, "defaults");
+            extract_option(options, "noauto");
+            if (noauto)
+                options.push("noauto");
+            if (options.length == 0)
+                options.push("defaults");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19843-20240116-090759-3ab49c70-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)